File: SDKs/microsoft.com/cognitiveservices-Training/2.1/go/pkg/models/operations/quicktestimage.go
package operations

import (
"openapi/pkg/models/shared")

type QuickTestImagePathParams struct {
    ProjectID string `pathParam:"style=simple,explode=false,name=projectId"`
    
}

type QuickTestImageQueryParams struct {
    IterationID *string `queryParam:"style=form,explode=true,name=iterationId"`
    
}

type QuickTestImageHeaders struct {
    TrainingKey string `header:"style=simple,explode=false,name=Training-Key"`
    
}

type QuickTestImageRequestBodyImageData struct {
    Content []byte `multipartForm:"content"`
    ImageData string `multipartForm:"name=imageData"`
    
}

type QuickTestImageRequestBody struct {
    ImageData QuickTestImageRequestBodyImageData `multipartForm:"file"`
    
}

type QuickTestImageRequest struct {
    PathParams QuickTestImagePathParams 
    QueryParams QuickTestImageQueryParams 
    Headers QuickTestImageHeaders 
    Request QuickTestImageRequestBody `request:"mediaType=multipart/form-data"`
    
}

type QuickTestImageResponse struct {
    Body []byte 
    ContentType string 
    ImagePrediction *shared.ImagePrediction 
    StatusCode int64 
    
}

File: SDKs/microsoft.com/cognitiveservices-Training/2.1/python/src/sdk/models/operations/quicktestimage.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class QuickTestImagePathParams:
    project_id: str = field(metadata={'path_param': { 'field_name': 'projectId', 'style': 'simple', 'explode': False }})
    

@dataclass
class QuickTestImageQueryParams:
    iteration_id: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'iterationId', 'style': 'form', 'explode': True }})
    

@dataclass
class QuickTestImageHeaders:
    training_key: str = field(metadata={'header': { 'field_name': 'Training-Key', 'style': 'simple', 'explode': False }})
    

@dataclass
class QuickTestImageRequestBodyImageData:
    content: bytes = field(metadata={'multipart_form': { 'content': True }})
    image_data: str = field(metadata={'multipart_form': { 'field_name': 'imageData' }})
    

@dataclass
class QuickTestImageRequestBody:
    image_data: QuickTestImageRequestBodyImageData = field(metadata={'multipart_form': { 'file': True }})
    

@dataclass
class QuickTestImageRequest:
    headers: QuickTestImageHeaders = field()
    path_params: QuickTestImagePathParams = field()
    query_params: QuickTestImageQueryParams = field()
    request: QuickTestImageRequestBody = field(metadata={'request': { 'media_type': 'multipart/form-data' }})
    

@dataclass
class QuickTestImageResponse:
    content_type: str = field()
    status_code: int = field()
    body: Optional[bytes] = field(default=None)
    image_prediction: Optional[shared.ImagePrediction] = field(default=None)
    
